lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -214,4 +223,112 @@ void LLDBMemoryReader::popLocalBuffer() {
   m_local_buffer_size = 0;
 }
 
+std::pair<uint64_t, uint64_t>
+LLDBMemoryReader::addModuleToAddressMap(ModuleSP module) {
+  // We use 0x2000000000000000 to tell that the address is one mapped by us when
+  // doing memory reads later on. This should not clash with any masks used by
+  // Swift.
+  uint64_t next_available_address = 0x2000000000000000;
+  if (!m_address_map.empty()) {
+    // We map the images contiguously one after the other, all with the highest
+    // bit set.
+    next_available_address = m_address_map.back().first;
+    // Add 8 to it to keep it pointer aligned. This assumes the previous address
+    // was also pointer aligned.
+    next_available_address = next_available_address + 8;
+  }
+
+#ifndef NDEBUG
+  static std::initializer_list<uint64_t> used_swift_masks = {
+      SWIFT_ABI_ARM_OBJC_WEAK_REFERENCE_MARKER_MASK,
+      SWIFT_ABI_ARM_OBJC_WEAK_REFERENCE_MARKER_VALUE,
+      SWIFT_ABI_ARM_IS_OBJC_BIT,
+      SWIFT_ABI_X86_64_SWIFT_SPARE_BITS_MASK,
+      SWIFT_ABI_X86_64_OBJC_RESERVED_BITS_MASK,
+      SWIFT_ABI_X86_64_SIMULATOR_OBJC_RESERVED_BITS_MASK,
+      SWIFT_ABI_X86_64_IS_OBJC_BIT,
+      SWIFT_ABI_X86_64_OBJC_WEAK_REFERENCE_MARKER_MASK,
+      SWIFT_ABI_X86_64_SIMULATOR_OBJC_WEAK_REFERENCE_MARKER_MASK,
+      SWIFT_ABI_X86_64_SIMULATOR_OBJC_WEAK_REFERENCE_MARKER_VALUE,
+      SWIFT_ABI_ARM64_SWIFT_SPARE_BITS_MASK,
+      SWIFT_ABI_ARM64_OBJC_RESERVED_BITS_MASK,
+      SWIFT_ABI_ARM64_IS_OBJC_BIT,
+      SWIFT_ABI_ARM64_OBJC_WEAK_REFERENCE_MARKER_MASK};
+
+  for (auto swift_mask : used_swift_masks) {
+    assert((next_available_address & swift_mask) != swift_mask &&
+           "LLDB artificial address clashes with a swift mask!");
```

Comment:
neat!
If we use static_assert, maybe this could even be a constexpr and be checked even if assertions are disabled.

Reply:
Unfortunately this can't can't be be a const_expr since `next_available_address` depends on the size of the images at runtime.

Reply:
IIUC, what we want to check is that the 0x2000000000000000 constant will never appear in a pointer that would appear in the process. For that we just need to check that it's contained in SWIFT_ABI_ARM64_SWIFT_SPARE_BITS_MASK.
So a `static_assert(0x2000000000000000 & SWIFT_ABI_ARM64_SWIFT_SPARE_BITS_MASK == 0x2000000000000000)` should be sufficient for that purpose. (Repeat with all other platforms).